pandas/core/arrays/timedeltas.py

```diff
@@ -128,25 +139,67 @@ def dtype(self):
     # Constructors
     _attributes = ["freq"]
 
-    @classmethod
-    def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
-        # `dtype` is passed by _shallow_copy in corner cases, should always
-        #  be timedelta64[ns] if present
-        assert dtype == _TD_DTYPE
-        assert isinstance(values, np.ndarray), type(values)
+    def __init__(self, values, dtype=_TD_DTYPE, freq=None, copy=False):
+        if isinstance(values, (ABCSeries, ABCIndexClass)):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            if freq is None:
+                freq = values.freq
+            elif freq and values.freq:
+                freq = to_offset(freq)
+                freq, freq_infer = dtl.validate_inferred_freq(
+                    freq, values.freq,
+                    freq_infer=False
+                )
+            values = values._data
+
+        if not isinstance(values, np.ndarray):
+            msg = (
+                "Unexpected type '{}'. 'values' must be a TimedeltaArray "
+                "ndarray, or Series or Index containing one of those."
+            )
+            raise ValueError(msg.format(type(values).__name__))
 
         if values.dtype == 'i8':
-            values = values.view('m8[ns]')
-
-        assert values.dtype == 'm8[ns]'
-
-        result = object.__new__(cls)
-        result._data = values
-        result._freq = freq
-        return result
+            # for compat with datetime/timedelta/period shared methods,
+            #  we can sometimes get here with int64 values.  These represent
+            #  nanosecond UTC (or tz-naive) unix timestamps
+            values = values.view(_TD_DTYPE)
+
+        if values.dtype != _TD_DTYPE:
+            msg = (
+                "The dtype of 'values' is incorrect. Must be "
+                "'timedelta64[ns]'."
+            )
+            raise ValueError(msg.format(values.dtype))
+
+        dtype_msg = "'dtype' must be 'timedelta64[ns]'. Got '{}' instead."
+        try:
+            if dtype != _TD_DTYPE:
+                raise ValueError(dtype_msg.format(dtype))
+        except TypeError:
+            raise ValueError(dtype_msg.format(dtype))
+
+        if freq == "infer":
+            msg = (
+                "Frequency inference not allowed in TimedeltaArray.__init__. "
+                "Use 'pd.array()' instead."
+            )
+            raise ValueError(msg)
+
+        if copy:
+            values = values.copy()
+        if freq:
+            freq = to_offset(freq)
+
+        self._data = values
+        self._dtype = dtype
+        self._freq = freq
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
-        return cls._from_sequence(values, dtype=dtype, copy=copy, freq=freq)
+    @classmethod
```

Comment:
can you define this in the super class (as I think it repeats fro DTA)? ahh, maybe not as you have a dtype= default

Reply:
They ended up quite similar. The default dtype isn't a deal breaker. We can just make that None.

The main additional complexity in DTA is the timezone handling. I suspect they could be merged with a bit of effort.